posthog/admin/admins/team_admin.py

```diff
@@ -242,3 +391,124 @@ def delete_cache(self, request, object_id):
+                for chunk in import_file.chunks():
+                    tmp_file.write(chunk)
+                tmp_file_path = tmp_file.name
+
+            temporal = sync_connect()
+            workflow_input = ImportRecordingInput(team_id=team.id, export_file=tmp_file_path)
+            workflow_id = f"import-recording-{team.id}-{uuid.uuid4()}"
+
+            asyncio.run(
+                temporal.start_workflow(
+                    "import-recording",
+                    workflow_input,
+                    id=workflow_id,
+                    task_queue=settings.SESSION_REPLAY_TASK_QUEUE,
+                    retry_policy=common.RetryPolicy(
+                        maximum_attempts=2,
+                        initial_interval=timedelta(minutes=1),
+                    ),
+                )
+            )
+
+            messages.success(
+                request,
+                f"Import triggered for team '{team.name}' by {request.user.email}.",
+            )
+        except Exception as e:
+            logger.exception(
+                "import_replay_failed",
+                team_id=team.id,
+                error=str(e),
+            )
+            messages.error(request, f"Import failed: {e}")
+
+        return redirect(reverse("admin:posthog_team_change", args=[object_id]))
+
+    def export_history_view(self, request, object_id):
+        team = Team.objects.get(pk=object_id)
+
+        exports = ExportedRecording.objects.filter(team=team).order_by("-created_at")[:50]
+
+        context = {
+            **self.admin_site.each_context(request),
+            "team": team,
+            "exports": exports,
+            "title": f"Export History - {team.name}",
+        }
+        return render(request, "admin/posthog/team/export_history.html", context)
+
+    def download_export_view(self, request, object_id, export_id):
+        from django.http import HttpResponse
+
+        from posthog.storage import session_recording_v2_object_storage
+
+        team = Team.objects.get(pk=object_id)
+        try:
+            export = ExportedRecording.objects.get(id=export_id, team=team)
+
+            if not export.export_location:
+                messages.error(request, "Export content not available yet")
+                return redirect(reverse("admin:posthog_team_export_history", args=[object_id]))
+
+            storage = session_recording_v2_object_storage.client()
+            content = storage.read_all_bytes(export.export_location)
+
+            if not content:
+                messages.error(request, "Failed to read export content from storage")
+                return redirect(reverse("admin:posthog_team_export_history", args=[object_id]))
+
+            response = HttpResponse(content, content_type="application/zip")
+            filename = f"export-{export.session_id}.zip"
+            response["Content-Disposition"] = f'attachment; filename="{filename}"'
```

Comment:
i bet we could do something clever here to stream the data from S3 to the response to avoid having to read very large files in to memory
(but then, if it works, we don't need clever)

Reply:
We can always just go and download the file from S3 using the AWS console if need be